### PR TITLE
Enrich CI: add static analyzers (slither for solidity), cache Python dependencies

### PR DESCRIPTION
**Pull request by achab**

### CODE REVIEW

**Review by hsouf — COMMENTED** [review]
How about outputting the number of detected vulnerabilities in a `known vulnerabilities` badge on the doc?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,26 +17,42 @@ env:
...
         run: |
           python -m venv .venv
           source .venv/bin/activate
           make
 
+      - name: Static analysis for Solidity
+        run: ./scripts/static_analysis_solidity.sh
+
+      - name: Static analysis for Cairo
+        run: python scripts/static_analysis_cairo.py
```
Comment (JorikSchellekens) [inline .github/workflows/ci.yml:54]: We should fail the ci if the sarif produced by the ci is different to the sarif checked into the repo

Comment (achab) [reply .github/workflows/ci.yml:54]: To ensure that we commit vulnerabilities files in each PR? I'm not sure yet it's a good idea to do so

Comment (achab) [reply .github/workflows/ci.yml:54]: We could add these in a pre-push hook, but these analyses take too much time to be run for each commit

**Review by hsouf — APPROVED** [review]
How about outputting the number of detected vulnerabilities in a `known vulnerabilities` badge on the doc?

**Pull request merged**